lib/routes/pos/transactions/pos_transactions_page.dart

```diff
@@ -55,73 +55,103 @@ class _PosTransactionsState extends State<_PosTransactionsPage> {
 
   @override
   Widget build(BuildContext context) {
-    return new Scaffold(
-        key: _scaffoldKey,
-        appBar: new AppBar(
-          iconTheme: theme.appBarIconTheme,
-          textTheme: theme.appBarTextTheme,
-          backgroundColor: Color.fromRGBO(5, 93, 235, 1.0),
-          leading: backBtn.BackButton(),
-          title: new Text(
-            _title,
-            style: theme.appBarTextStyle,
-          ),
-          actions: <Widget>[
-            new Padding(
-              padding: EdgeInsets.only(left: 12.0, right: 0.0),
-              child: IconButton(icon: ImageIcon(
-                AssetImage("src/icon/calendar.png"),
-                color: Colors.white,
-                size: 24.0,
-              ),
-                onPressed: () =>
-                    showDialog(barrierDismissible: false,
-                      context: context,
-                      builder: (_) => CalendarDialog(context, _firstDate),),),
-            ),
-          ],
-          elevation: 0.0,
-        ),
-        body: StreamBuilder<AccountModel>(
+    return StreamBuilder<AccountModel>(
             stream: widget._accountBloc.accountStream,
             builder: (context, snapshot) {
               AccountModel account = snapshot.data;
               return StreamBuilder<PaymentsModel>(
                   stream: widget._accountBloc.paymentsStream,
                   builder: (context, snapshot) {
-                    List<PaymentInfo> payments = new List<PaymentInfo>();
+                    PaymentsModel paymentsModel = new PaymentsModel.initial();
```

Comment:
Don't create models in the UI, they are only created in the bloc and retrieved in the stream. If you get null from the stream it has meaning (the bloc hasn't been initialized yet)